lib/ansible/module_utils/vmware.py

```diff
@@ -432,6 +435,60 @@ def list_snapshots(vm):
     return result
 
 
+def get_vnc_extraconfig(vm):
+    result = {}
+    extraconfig = vm.config.extraConfig
+    for opts in extraconfig:
+        for optkeyname in ['enabled', 'ip', 'port', 'password']:
+            if opts.key.lower() == "remotedisplay.vnc." + optkeyname:
+                result[optkeyname] = opts.value
+    return result
+
+
+def set_vnc_extraconfig(content, vm, enabled, ip, port, password):
+    result = dict(
+        changed=False,
+        failed=False,
+    )
+    # set new values
+    key_prefix = "remotedisplay.vnc."
+    new_values = dict()
+    for key in ['enabled', 'ip', 'port', 'password']:
+        new_values[key_prefix + key] = ""
+    if enabled:
+        new_values[key_prefix + "enabled"] = "true"
+        new_values[key_prefix + "password"] = str(password).strip()
+        new_values[key_prefix + "ip"] = str(ip).strip()
+        new_values[key_prefix + "port"] = str(port).strip()
+    # get current vnc config
+    current_values = get_vnc_extraconfig(vm)
+    # check if any value is changed
+    reconfig_vm = False
+    for key, val in new_values.items():
+        key = key.replace(key_prefix, "")
+        if current_values.get(key, "") != val:
```

Comment:
I would write this as

```
if current_values.get(key, "").lower() != val.lower():
```

Reply:
@Akasurde yes, that would be good for enabled, but not for case-sensitive password. maybe something like this?
```
reconfig_vm = False
if current_values.get("password", "") != new_values[key_prefix + "password"]:
    reconfig_vm = True
else:
    for key, val in new_values.items():
        key = key.replace(key_prefix, "")
        if current_values.get(key, "").lower() != val.lower():
            reconfig_vm = True
if not reconfig_vm:
    return result
```

Reply:
How about this - 

```
    reconfig_vm = False

    for key, val in new_values.items():
        key = key.replace(key_prefix, "")
        current_value = current_values.get(key, '')
        if key == 'enabled':
            current_value = current_value.lower()
            val = val.lower()
        if current_value != val:
            reconfig_vm = True

    if not reconfig_vm:
        return result

```

Reply:
I am OK with either of the solution.